### PR TITLE
(chore) Add ESLint Rules for React-Hooks

### PR DESCRIPTION
**Pull request by nravilla**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [x] My work includes tests or is validated by existing tests.

## Summary
Description:
This pull request introduces the following updates to the OHRI repository:
**1. ESLint Rules for React-Hooks:**
- Implemented ESLint rules specific to React Hooks (missing dependencies, exhaustive-deps, import type) to ensure consistency and adherence to best practices.
- The rules are aligned with those used in the OpenMRS patient-chart project, promoting uniformity across projects.

**2. Husky and Lint-Staged Integration:**
- Added Husky to manage Git hooks, enabling automated tasks to run at various stages of the Git workflow.
- Configured Lint-Staged to run ESLint checks on staged files before commits, ensuring that code adheres to linting standards before being committed.

## Screenshots
image

image

## Other
The changes were tested by comparing localhost functionality with the OHRI-dev

### CODE REVIEW

**Comment by hadijahkyampeire** [comment]
@nravilla you have some merge conflicts, would you resolve them please.

**Review by eudson — COMMENTED** [review]
@nravilla the PR has conflicts

File: packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx [hunk 1]
```diff
@@ -100,48 +134,24 @@ const CurrentPregnancy: React.FC<PatientChartProps> = ({ patientUuid, pTrackerId
...
-  }, [relatives]);
+      return pTrackerMap;
+    },
+    [encounterTypes.PTrackerIdentifierType],
+  );
 
-  const getInfantOutcome = () => {
-    const infantOutcomes = relatives.map(async (relative) => {
+  const getInfantOutcome = useCallback(() => {
+    const infantOutcomesPromises = relatives.map(async (relative) => {
```
Comment (hadijahkyampeire) [inline packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx:154]: why change this name

File: yarn.lock [hunk 2]
```diff
@@ -6841,9 +6841,9 @@ __metadata:
   linkType: hard
 
 "caniuse-lite@npm:^1.0.0, caniuse-lite@npm:^1.0.30001400, caniuse-lite@npm:^1.0.30001426":
-  version: 1.0.30001517
-  resolution: "caniuse-lite@npm:1.0.30001517"
-  checksum: 3b9aa01e5e1c112bbe15d653cf4f909748b3ede87613b4381306c3510dab1adb0b09594febdacdf0eb963d8f0d37dcb3b88cea12c579ed14a9ede4581544c7c4
+  version: 1.0.30001632
```
Comment (hadijahkyampeire) [inline yarn.lock:6844]: We should restrain from modifying the yarn.lock unless we add a package.

**Review by hadijahkyampeire — COMMENTED** [review]
Thanks @nravilla just a few comments

**Comment by nravilla** [comment]
Hi Hadijah, Eudson,

thank you for your excellent catch (commented line and yarn.lock) during review yesterday. i have resolved all the issues that was raised.  please let me know if you have any questions or concerns, Thank you so much!

**Review by hadijahkyampeire — APPROVED** [review]
Thanks @nravilla we can merge this.

**Pull request merged**